### PR TITLE
[PPP-10169] Setup Initial Schemas

### PR DESCRIPTION
**Pull request by jlam-mpc**
Features:
- Setup Forecast table
- Setup all reference tables used by Forecast objects
- Setup HighlightMessage table
- Setup Announcement table

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -1,17 +1,261 @@
...
+    previewFeatures = ["fullTextSearch", "fullTextIndex"]
 }
 
 datasource db {
-    provider = "mysql"
-    url      = env("DATABASE_URL")
+    provider     = "mysql"
+    url          = env("DATABASE_URL")
+    // It is possible we may switch to foreignKeys relationmode if stop using PlanetScale
+    relationMode = "prisma"
```
Comment (scott062) [inline prisma/schema.prisma:15]: This seems to be a trivial change for our code, but will hopefully have a performance boost in AWS. Most likely will cutover from PlanetScale to AWS after proof-of-concept for local dev

Comment (jlam-mpc) [reply prisma/schema.prisma:15]: The only issue is, once we change it it won't work with PlanetScale so all devs who have it connected to PS would need to change that.

Comment (jlam-mpc) [reply prisma/schema.prisma:15]: Changing this after POC makes sense

File: prisma/schema.prisma [hunk 2]
```diff
@@ -1,17 +1,261 @@
...
+    active          Boolean   @default(true)
+    featured        Boolean   @default(false)
+    state           Int?
+
+    // Flat Attributes
+    anticipated_award_date                DateTime?
+    anticipated_solicitation_release_date DateTime?
+    co_contact                            String?
+    dollar_value_with_base_option         String?
+    fiscal_year                           Int?
```
Comment (elizabeth-jimenez) [inline prisma/schema.prisma:77]: Ah, this is `int` bc i'm assuming it's one of those iffy three? I see the commented out portion in the seed

**Comment by elizabeth-jimenez** [comment]
HighlightMessage Table
I see `number` and `state` in json but not in schema.
Data Questions: 
Are we expecting updated_by and created_by values? or will they be PPP generated values?
What are work_notes and work_notes_list?
What are number and state in json file?
Do we want to default active to True?

**Comment by elizabeth-jimenez** [comment]
Announcement Table
`glyp` and `public` in json but not in schema
Data Questions:
What are `glyph` and `public`?

**Comment by elizabeth-jimenez** [comment]
Forecast Table
seeing `u_archive_opportunity` in json but not in schema

Data Questions:
What are `opened` and `opened_by`? Are those gonna be PPP data values?
Seeing `anticipated_award_date`, `anticipated_solicitation_release_date`, `co_contact`, `dollar_value_with_base_option`, and `long_description` in schema but not json.

**Comment by elizabeth-jimenez** [comment]
Sorry if my data questions are ignorant, this is the most familiar i've gotten with the data 😅

File: prisma/schema.prisma (hunk 2 above)
Comment (jlam-mpc) [reply prisma/schema.prisma:77]: Rightfully it should be a constrained `int` in the db only accepting values 1-4 but currently my seeding function doesn't know how to compile Number() and charAt() which is still being investigated.

**Comment by jlam-mpc** [comment]
> **HighlightMessage Table** I see `number` and `state` in json but not in schema.
> 
> Data Questions: Are we expecting updated_by and created_by values? or will they be PPP generated values? What are work_notes and work_notes_list? What are number and state in json file? Do we want to default active to True?

`number` is translated to `listing_id` in our db which is the gov't specific ID for each forecast object
`state` is already in there under Forecasts (beneath `featured`)
It's something that SN uses to denote what state it's in but I'm currently unsure what logic it's used under

I think we may not have to worry about updated_by/created_by/work_notes/etc. but I left them as a part of our architecture in case the client one day wants more SN data into our portal.

**Comment by jlam-mpc** [comment]
> 

`glyph` may be a SN specific field but I can double check with Emily. As for `public` I'm not sure what it's used for but I can go ahead and add it in.

**Comment by jlam-mpc** [comment]
> **Forecast Table** seeing `u_archive_opportunity` in json but not in schema
> 
> Data Questions: What are `opened` and `opened_by`? Are those gonna be PPP data values? Seeing `anticipated_award_date`, `anticipated_solicitation_release_date`, `co_contact`, `dollar_value_with_base_option`, and `long_description` in schema but not json.

I'll go ahead and ahh an `archive` field as well, I just wasn't 100% sure how archive, active, and state all differentiate from one another but let me clear that up with Emily.

`opened` and `opened_by` are not PPP data values. They're from SN and we probably don't have to display them anywhere but again, just keeping it in the design to stay consistent with SN backend and allow for any changes in the future to include them.

All of the fields you listed are in the Forecast table under "Flat Attributes"

**Review by elizabeth-jimenez — APPROVED** [review]
🥳

**Comment by elizabeth-jimenez** [comment]
Thanks for answering my million questions :)

**Pull request merged**